realm/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -34,31 +30,60 @@
  */
 public class DynamicRealmObject extends RealmObject {
 
-     Realm realm;
-     Row row;
-
     /**
      * Creates a dynamic Realm object based on a existing object.
      *
      * @param obj Realm object to convert to a dynamic object. Only objects managed by Realm can be used.
-     * @throws IllegalArgumentException if object isn't managed by a Realm.
+     * @throws IllegalArgumentException if object isn't managed by Realm.
      */
     public DynamicRealmObject(RealmObject obj) {
         if (obj == null) {
             throw new IllegalArgumentException("Non-null object must be provided.");
         }
-        Row row = RealmObject.getRow(obj);
+        Row row = obj.row;
         if (row == null) {
-            throw new IllegalArgumentException("A object managed by Realm must be provided. This is a standalone object.");
+            throw new IllegalArgumentException("A object managed by Realm must be provided. This " +
+                    "is a standalone object.");
         }
-        this.realm = RealmObject.getRealm(obj);
-        this.row = (row instanceof CheckedRow) ? (CheckedRow) row : ((UncheckedRow) row).convertToChecked();
+        this.realm = obj.realm;
+        this.row = (row instanceof CheckedRow) ? row : ((UncheckedRow) row).convertToChecked();
     }
 
     // Create a dynamic object. Only used internally
-    DynamicRealmObject(Realm realm, CheckedRow row) {
+    DynamicRealmObject() {
+
+    }
+
+    DynamicRealmObject(BaseRealm realm, Row row) {
         this.realm = realm;
-        this.row = row;
+        this.row = (row instanceof CheckedRow) ? (CheckedRow) row : ((UncheckedRow) row).convertToChecked();
```

Comment:
Because `DynamicRealmObject extends RealmObject` it is actually possible to do ```new DynamicRealmObject(otherDynamicObject)``` I don't think there is as valid use case for it though, so I should probably just throw an exception in that case.